acceptance/acceptance_test.go

```diff
@@ -146,6 +155,58 @@ func sharedCmds(tsEnv testScriptEnv) map[string]func(ts *testscript.TestScript,
 				}
 			})
 		},
+		"env2upper": func(ts *testscript.TestScript, neg bool, args []string) {
+			if neg {
+				ts.Fatalf("unsupported: ! env2upper")
+			}
+			if len(args) == 0 {
+				ts.Fatalf("usage: env2upper name=value ...")
+			}
+			for _, env := range args {
+				i := strings.Index(env, "=")
+
+				if i < 0 {
+					ts.Fatalf("env2upper: argument does not match name=value")
+				}
+
+				ts.Setenv(env[:i], strings.ToUpper(env[i+1:]))
+			}
+		},
+		"replace": func(ts *testscript.TestScript, neg bool, args []string) {
+			if neg {
+				ts.Fatalf("unsupported: ! replace")
+			}
+			if len(args) < 2 {
+				ts.Fatalf("usage: replace file env...")
+			}
+
+			src := ts.MkAbs(args[0])
+			ts.Logf("replace src: %s", src)
+			info, err := os.Stat(src)
+			ts.Check(err)
+			mode := info.Mode() & 0o777
```

Comment:
I think the idea here is that we always write the file with the same permissions it already had? If so, maybe you can write a comment explaining that.